pkg/platform/packet/packet.go

```diff
@@ -288,6 +294,46 @@ func createTerraformConfigFile(cfg *config, terraformPath string) error {
 	return nil
 }
 
+func validateNodePrivateCIDRs(cfg *config) error {
+	// If both the deprecated and the new fields are provided then error out.
+	if cfg.NodePrivateCIDR != "" && len(cfg.NodePrivateCIDRs) > 0 {
+		//nolint:stylecheck
+		return fmt.Errorf("`node_private_cidr` and `node_private_cidrs` cannot be used together." +
+			" Use `node_private_cidrs` only.")
+	}
+
+	// If just the deprecated field is provided then produce a warning.
+	if cfg.NodePrivateCIDR != "" {
+		fmt.Printf("Warning: `node_private_cidr` is deprecated, use `node_private_cidrs` instead.\n")
+
+		cfg.NodePrivateCIDRs = []string{cfg.NodePrivateCIDR}
+
+		// createTerraformConfigFile is called multiple times (once during the `terraform init` and
+		// during `terraform apply`), in turn this function is called multiple times. If we make
+		// changes to the object `cfg` once it is fine but when called again and if we make
+		// modifications then these checks give wrong results. If a user sets only
+		// `node_private_cidr` and here we save it to `NodePrivateCIDRs` and if this function is
+		// called twice then we have data in two places in `NodePrivateCIDR` and in
+		// `NodePrivateCIDRs` so this function fails the second time at the first condition of this
+		// function.
+		cfg.NodePrivateCIDR = ""
```

Comment:
When you run `lokoctl cluster apply`

### Call stack 1

```go
func Apply(contextLogger *log.Entry, options ApplyOptions) error {
```
:arrow_down:
```go
func (c *config) Apply(ex *terraform.Executor) error {
```
:arrow_down: 
```go
func (c *config) Initialize(ex *terraform.Executor) error {
```
:arrow_down:
```go
func createTerraformConfigFile(cfg *config, terraformPath string) error {
```
:arrow_down:
```go
func validateNodePrivateCIDRs(cfg *config) error {
```

### Call stack 2

```go
func Apply(contextLogger *log.Entry, options ApplyOptions) error {
```
:arrow_down:
```go
func (cc clusterConfig) initialize(contextLogger *log.Entry) (*cluster, error) {
```
:arrow_down:
```go
func (cc clusterConfig) initializeTerraform(p platform.Platform, b backend) (*terraform.Executor, error) {
```
:arrow_down:
```go
func (c *config) Initialize(ex *terraform.Executor) error {
```
:arrow_down:
```go
func createTerraformConfigFile(cfg *config, terraformPath string) error {
```
:arrow_down:
```go
func validateNodePrivateCIDRs(cfg *config) error {
```